File: 1521-find-a-value-of-a-mysterious-function-closest-to-target/1521-find-a-value-of-a-mysterious-function-closest-to-target.py
class Solution(object):
    def closestToTarget(self, arr, target):
        """
        :type arr: List[int]
        :type target: int
        :rtype: int
        """
        s = set()
        res = float('inf')
        for i in range(len(arr)):
            s1 = set()
            for x in s:
                s1.add(x & arr[i])
            s1.add(arr[i])
            for x in s1:
                res = min(res, abs(x - target))
            s = s1
        return res

File: 1521-find-a-value-of-a-mysterious-function-closest-to-target/1521-find-a-value-of-a-mysterious-function-closest-to-target.cpp
class Solution {
public:
    int closestToTarget(vector<int>& arr, int target) {
        unordered_set<int> s;
        int res = INT_MAX;
        for(int i = 0; i < arr.size(); ++i) {
            unordered_set<int> s1;
            for(int x : s) s1.insert(x & arr[i]);
            s1.insert(arr[i]);
            for(int x : s1) res = min(res, abs(x - target));
            s = s1;
        }
        return res;
    }
};